lib/discourse_webauthn/authentication_service.rb

```diff
@@ -33,7 +33,7 @@ def authenticate_security_key
       #     verify that response.userHandle is present. Verify that the user account identified by response.userHandle
       #     contains a credential record whose id equals credential.rawId
       if @factor_type == UserSecurityKey.factor_types[:first_factor] &&
-           Base64.decode64(@params[:userHandle]) != @current_user.secure_identifier
+           Base64.decode64(@params[:userHandle]) != security_key.user.secure_identifier
```

Comment:
This is a small bug fix, introduced in https://github.com/discourse/discourse/commit/921f1279b9a35be14ae619ceed4a20127425df57 (but without routes or UI, this wasn't used anywhere).